<!DOCTYPE html>
<html>
<head>
    <script defer src="https://umami.dudleya.cc/script.js" data-website-id="ced50e3a-a449-456e-ac66-f7dfe60706c9"></script>

    <title>NGS_lab1 // Meteor03&#39;s Oxalis Garden</title>

    <meta charset="utf-8">
    <meta http-equiv="content-type" content="text/html; charset=utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">
    
    
    

        <meta property="og:title" content="NGS_lab1" />
    <meta property="og:description" content="" />
    <meta property="og:type" content="website" />
    <meta property="og:locale" content="en" />
    <meta property="og:url" content="https://meteor-oxalis3.github.io/ngs/ngs_lab1/" />
    

    <link rel="shortcut icon" href="/favicon.ico">

    <link href="https://meteor-oxalis3.github.io/webfonts/ptserif/main.css" rel='stylesheet' type='text/css'>
    <link href="https://meteor-oxalis3.github.io/webfonts/source-code-pro/main.css" rel="stylesheet" type="text/css">

    <link rel="stylesheet" href="https://meteor-oxalis3.github.io/css/style.css">
    

    <meta name="generator" content="Hugo 0.111.3">
</head>


<body>
<div id="container">
    <header id="header">
    <div id="header-outer" class="outer">
        <div id="header-inner" class="inner">
            <a id="main-nav-toggle" class="nav-icon" href="javascript:;"></a>
            <a id="logo" class="logo-text" href="https://meteor-oxalis3.github.io/">Meteor03&#39;s Oxalis Garden</a>
            <nav id="main-nav">
                
                <a class="main-nav-link" href="/linux/">Linux</a>
                
                <a class="main-nav-link" href="/ngs/">NGS</a>
                
                <a class="main-nav-link" href="">Microarray</a>
                
                <a class="main-nav-link" href="/article/">Article</a>
                
            </nav>
            <nav id="sub-nav">
                <div id="search-form-wrap">
                </div>
            </nav>
        </div>
    </div>
</header>

    <section id="main" class="outer">
        <article class="article article-type-post" itemscope itemprop="blogPost">
    <div class="article-inner">
        
        <header class="article-header">
            <h1 class="article-title" itemprop="name">NGS_lab1</h1>
        </header>
        
        <div class="article-meta">
            <a href="/ngs/ngs_lab1/" class="article-date">
                <time datetime='2024-10-11T09:31:45.000&#43;08:00' itemprop="datePublished">2024-10-11</time>
            </a>
            
            
            
        </div>
        <div class="article-entry" itemprop="articleBody">
            <h1 id="æ•°æ®ä¸‹è½½ä¸è´¨é‡è¯„ä¼°">æ•°æ®ä¸‹è½½ä¸è´¨é‡è¯„ä¼°</h1>
<h2 id="æ•°æ®åº“ç½‘ç«™">æ•°æ®åº“ç½‘ç«™</h2>
<ul>
<li><a href="https://www.ncbi.nlm.nih.gov/sra">SRA (Sequence Read Archive)</a></li>
<li><a href="https://www.ebi.ac.uk/ena/browser/home">ENA (European Nucleotide Archive)</a></li>
<li><a href="https://ngdc.cncb.ac.cn/gsa/">GSA (Genome Sequence Archive)</a></li>
</ul>
<hr>
<h2 id="æ•°æ®ä¸‹è½½å·¥å…·">æ•°æ®ä¸‹è½½å·¥å…·</h2>
<ul>
<li><a href="https://github.com/ncbi/sra-tools/wiki/01.-Downloading-SRA-Toolkit">SRA Toolkit</a></li>
<li><a href="https://www.gnu.org/software/wget/">Wget</a></li>
<li><a href="https://aspera.ibmaspera.com/help/aspera_connect">Aspera Connect</a></li>
</ul>
<hr>
<h2 id="è´¨é‡è¯„ä¼°å·¥å…·">è´¨é‡è¯„ä¼°å·¥å…·</h2>
<ul>
<li><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a></li>
<li><a href="https://multiqc.info/">MultiQC</a></li>
</ul>
<hr>
<h2 id="å®‰è£…-sra-toolkit">å®‰è£… <code>SRA Toolkit</code></h2>
<p>( æŒ‘ä¸ªä½ å–œæ¬¢çš„æ–¹æ³•å®‰è£…å°±å¥½ï½ )</p>
<ul>
<li>æ–¹æ³•ä¸€ï¼š<code>anaconda</code> æˆ– <code>micromamba</code> (<strong>æ¨è</strong>)</li>
<li>æ–¹æ³•äºŒï¼šä¸‹è½½é¢„ç¼–è¯‘çš„äºŒè¿›åˆ¶æ–‡ä»¶</li>
</ul>
<h3 id="ä½¿ç”¨-anaconda-å®‰è£…">ä½¿ç”¨ <code>anaconda</code> å®‰è£…</h3>
<ul>
<li>æ–°å»ºè™šæ‹Ÿç¯å¢ƒ <code>sra-tools</code></li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span>conda create -n sra-tools
</span></span><span style="display:flex;"><span><span style="color:#75715e"># ç­‰å¾…ä¸€ä¼šå„¿åè¾“å…¥ y , ç„¶åå›è½¦</span>
</span></span></code></pre></div><ul>
<li>è¿›å…¥è™šæ‹Ÿç¯å¢ƒ <code>sra-tools</code></li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span>conda activate sra-tools
</span></span></code></pre></div><ul>
<li>å®‰è£… <code>SRA Toolkit</code></li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span><span style="color:#75715e"># æ³¨æ„ï¼å¦‚æœä¸æ·»åŠ  -c conda-forge, åˆ™é»˜è®¤æ— æ³•ä¸‹è½½æœ€æ–°ç‰ˆæœ¬çš„ sra-tools</span>
</span></span><span style="display:flex;"><span>conda install -c bioconda -c conda-forge sra-tools
</span></span><span style="display:flex;"><span><span style="color:#75715e"># ç­‰å¾…ä¸€ä¼šå„¿åè¾“å…¥ y , ç„¶åå›è½¦</span>
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æ£€éªŒæ˜¯å¦å®‰è£…æˆåŠŸ</span>
</span></span><span style="display:flex;"><span>srapath --version
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æˆ‘çš„è¾“å‡ºæ˜¯: </span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># srapath : 3.1.1</span>
</span></span></code></pre></div><h3 id="ä½¿ç”¨-micromamba-å®‰è£…">ä½¿ç”¨ <code>micromamba</code> å®‰è£…</h3>
<ul>
<li>æ–°å»ºè™šæ‹Ÿç¯å¢ƒ <code>sra-tools</code></li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span><span style="color:#75715e"># å’Œ anaconda ä¸åŒ, micromamba åœ¨æ–°å»ºè™šæ‹Ÿç¯å¢ƒçš„æ—¶å€™éœ€è¦åŠ ä¸Š env</span>
</span></span><span style="display:flex;"><span>micromamba env create -n sra-tools
</span></span></code></pre></div><ul>
<li>è¿›å…¥è™šæ‹Ÿç¯å¢ƒ <code>sra-tools</code></li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span>micromamba activate sra-tools
</span></span></code></pre></div><ul>
<li>å®‰è£… <code>SRA Toolkit</code></li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span><span style="color:#75715e"># æ³¨æ„ï¼å¦‚æœä¸æ·»åŠ  -c conda-forge, åˆ™é»˜è®¤æ— æ³•ä¸‹è½½æœ€æ–°ç‰ˆæœ¬çš„ sra-tools</span>
</span></span><span style="display:flex;"><span>micromamba install -c bioconda -c conda-forge sra-tools
</span></span><span style="display:flex;"><span><span style="color:#75715e"># ç­‰å¾…ä¸€ä¼šå„¿åè¾“å…¥ Y , ç„¶åå›è½¦</span>
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æ£€éªŒæ˜¯å¦å®‰è£…æˆåŠŸ</span>
</span></span><span style="display:flex;"><span>srapath --version
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æˆ‘çš„è¾“å‡ºæ˜¯: </span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># srapath : 3.1.1</span>
</span></span></code></pre></div><hr>
<h2 id="å®‰è£…-fastqc">å®‰è£… <code>FastQC</code></h2>
<p>( æŒ‘ä¸ªä½ å–œæ¬¢çš„æ–¹æ³•å®‰è£…å°±å¥½ï½ )</p>
<ul>
<li>æ–¹æ³•ä¸€ï¼š<code>anaconda</code> æˆ– <code>micromamba</code> (<strong>æ¨è</strong>)</li>
<li>æ–¹æ³•äºŒï¼šä¸‹è½½é¢„ç¼–è¯‘çš„äºŒè¿›åˆ¶æ–‡ä»¶</li>
</ul>
<h3 id="ä½¿ç”¨-anaconda-å®‰è£…-1">ä½¿ç”¨ <code>anaconda</code> å®‰è£…</h3>
<ul>
<li>æ–°å»ºè™šæ‹Ÿç¯å¢ƒ <code>fastqc</code></li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span>conda create -n fastqc
</span></span><span style="display:flex;"><span><span style="color:#75715e"># ç­‰å¾…ä¸€ä¼šå„¿åè¾“å…¥ y , ç„¶åå›è½¦</span>
</span></span></code></pre></div><ul>
<li>è¿›å…¥è™šæ‹Ÿç¯å¢ƒ <code>fastqc</code></li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span>conda activate fastqc
</span></span></code></pre></div><ul>
<li>å®‰è£… <code>FastQC</code></li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span>conda install -c bioconda fastqc
</span></span><span style="display:flex;"><span><span style="color:#75715e"># ç­‰å¾…ä¸€ä¼šå„¿åè¾“å…¥ y , ç„¶åå›è½¦</span>
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æ£€éªŒæ˜¯å¦å®‰è£…æˆåŠŸ</span>
</span></span><span style="display:flex;"><span>fastqc --version
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æˆ‘çš„è¾“å‡ºæ˜¯: </span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># FastQC v0.12.1</span>
</span></span></code></pre></div><h3 id="ä½¿ç”¨-micromamba-å®‰è£…-1">ä½¿ç”¨ <code>micromamba</code> å®‰è£…</h3>
<ul>
<li>æ–°å»ºè™šæ‹Ÿç¯å¢ƒ <code>fastqc</code></li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span><span style="color:#75715e"># å’Œ anaconda ä¸åŒ, micromamba åœ¨æ–°å»ºè™šæ‹Ÿç¯å¢ƒçš„æ—¶å€™éœ€è¦åŠ ä¸Š env</span>
</span></span><span style="display:flex;"><span>micromamba env create -n fastqc
</span></span></code></pre></div><ul>
<li>è¿›å…¥è™šæ‹Ÿç¯å¢ƒ <code>fastqc</code></li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span>micromamba activate fastqc
</span></span></code></pre></div><ul>
<li>å®‰è£… <code>FastQC</code></li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span><span style="color:#75715e"># æ³¨æ„ï¼å¦‚æœä¸æ·»åŠ  -c conda-forge, åˆ™é»˜è®¤æ— æ³•ä¸‹è½½æœ€æ–°ç‰ˆæœ¬çš„ sra-tools</span>
</span></span><span style="display:flex;"><span>micromamba install -c bioconda -c conda-forge fastqc
</span></span><span style="display:flex;"><span><span style="color:#75715e"># ç­‰å¾…ä¸€ä¼šå„¿åè¾“å…¥ Y , ç„¶åå›è½¦</span>
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æ£€éªŒæ˜¯å¦å®‰è£…æˆåŠŸ</span>
</span></span><span style="display:flex;"><span>fastqc --version
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æˆ‘çš„è¾“å‡ºæ˜¯: </span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># FastQC v0.12.1</span>
</span></span></code></pre></div><hr>
<h2 id="å®‰è£…-multiqc">å®‰è£… <code>MultiQC</code></h2>
<p>( æŒ‘ä¸ªä½ å–œæ¬¢çš„æ–¹æ³•å®‰è£…å°±å¥½ï½ )</p>
<ul>
<li>æ–¹æ³•ä¸€ï¼š<code>anaconda</code> æˆ– <code>micromamba</code> (<strong>æ¨è</strong>)</li>
<li>æ–¹æ³•äºŒï¼šä¸‹è½½é¢„ç¼–è¯‘çš„äºŒè¿›åˆ¶æ–‡ä»¶</li>
</ul>
<h3 id="ä½¿ç”¨-anaconda-å®‰è£…-2">ä½¿ç”¨ <code>anaconda</code> å®‰è£…</h3>
<ul>
<li>æ–°å»ºè™šæ‹Ÿç¯å¢ƒ <code>multiqc</code></li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span>conda create -n multiqc
</span></span><span style="display:flex;"><span><span style="color:#75715e"># ç­‰å¾…ä¸€ä¼šå„¿åè¾“å…¥ y , ç„¶åå›è½¦</span>
</span></span></code></pre></div><ul>
<li>è¿›å…¥è™šæ‹Ÿç¯å¢ƒ <code>multiqc</code></li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span>conda activate multiqc
</span></span></code></pre></div><ul>
<li>å®‰è£… <code>MultiQC</code></li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span><span style="color:#75715e"># æ³¨æ„ï¼å¦‚æœä¸æ·»åŠ  -c conda-forge, åˆ™é»˜è®¤æ— æ³•ä¸‹è½½æœ€æ–°ç‰ˆæœ¬çš„ multiqc</span>
</span></span><span style="display:flex;"><span>conda install -c bioconda -c conda-forge multiqc
</span></span><span style="display:flex;"><span><span style="color:#75715e"># ç­‰å¾…ä¸€ä¼šå„¿åè¾“å…¥ y , ç„¶åå›è½¦</span>
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æ£€éªŒæ˜¯å¦å®‰è£…æˆåŠŸ</span>
</span></span><span style="display:flex;"><span>multiqc --version
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æˆ‘çš„è¾“å‡ºæ˜¯: </span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># multiqc, version 1.8</span>
</span></span></code></pre></div><h3 id="ä½¿ç”¨-micromamba-å®‰è£…-2">ä½¿ç”¨ <code>micromamba</code> å®‰è£…</h3>
<ul>
<li>æ–°å»ºè™šæ‹Ÿç¯å¢ƒ <code>multiqc</code></li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span><span style="color:#75715e"># å’Œ anaconda ä¸åŒ, micromamba åœ¨æ–°å»ºè™šæ‹Ÿç¯å¢ƒçš„æ—¶å€™éœ€è¦åŠ ä¸Š env</span>
</span></span><span style="display:flex;"><span>micromamba env create -n multiqc
</span></span></code></pre></div><ul>
<li>è¿›å…¥è™šæ‹Ÿç¯å¢ƒ <code>multiqc</code></li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span>micromamba activate multiqc
</span></span></code></pre></div><ul>
<li>å®‰è£… <code>MultiQC</code></li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span><span style="color:#75715e"># æ³¨æ„ï¼å¦‚æœä¸æ·»åŠ  -c conda-forge, åˆ™é»˜è®¤æ— æ³•ä¸‹è½½æœ€æ–°ç‰ˆæœ¬çš„ multiqc</span>
</span></span><span style="display:flex;"><span>micromamba install -c bioconda -c conda-forge multiqc
</span></span><span style="display:flex;"><span><span style="color:#75715e"># ç­‰å¾…ä¸€ä¼šå„¿åè¾“å…¥ Y , ç„¶åå›è½¦</span>
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æ£€éªŒæ˜¯å¦å®‰è£…æˆåŠŸ</span>
</span></span><span style="display:flex;"><span>multiqc --version
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æˆ‘çš„è¾“å‡ºæ˜¯: </span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># multiqc, version 1.8</span>
</span></span></code></pre></div><hr>
<h2 id="ä½¿ç”¨æ•™ç¨‹">ä½¿ç”¨æ•™ç¨‹</h2>
<h2 id="sra-toolkit"><code>SRA Toolkit</code></h2>
<h3 id="ä¸‹è½½-sra-æ–‡ä»¶">ä¸‹è½½ <code>.sra</code> æ–‡ä»¶</h3>
<p>(å‡å¦‚æˆ‘æƒ³ä¸‹è½½ <code>SRR1646444</code>)</p>
<h4 id="æ–¹æ³•ä¸€-prefetch">æ–¹æ³•ä¸€ (prefetch)</h4>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span><span style="color:#75715e"># prefetch è‡ªå¸¦æ–­ç‚¹ç»­ä¼ </span>
</span></span><span style="display:flex;"><span>prefetch SRR1646444
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æ–‡ä»¶ SRR1646444.sra ä¿å­˜åœ¨å½“å‰ç›®å½•çš„ SRR1646444 æ–‡ä»¶å¤¹é‡Œé¢</span>
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># ä¼˜ç‚¹: prefetch åé¢å¯ä»¥åŠ ä¸Šå¤šä¸ª SRRå·, åˆ©äºä¸‹è½½å¤šä¸ª .sra æ–‡ä»¶</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># ç¼ºç‚¹: æ²¡æœ‰è¿›åº¦æ¡, ä¸æ–¹ä¾¿æŸ¥çœ‹ä¸‹è½½è¿›åº¦</span>
</span></span></code></pre></div><h4 id="æ–¹æ³•äºŒ-wget--srapath">æ–¹æ³•äºŒ (wget + srapath)</h4>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span><span style="color:#75715e"># srapath è·å–çš„æ˜¯ .sra æ–‡ä»¶å¯¹åº”çš„ç½‘å€</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># -c å‚æ•°ä½¿å¾— wget æ”¯æŒæ–­ç‚¹ç»­ä¼ </span>
</span></span><span style="display:flex;"><span>wget -c <span style="color:#66d9ef">$(</span>srapath SRR1646444<span style="color:#66d9ef">)</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æ–‡ä»¶ SRR1646444 ä¿å­˜åœ¨å½“å‰ç›®å½• (æ²¡æœ‰ .sra åç¼€)</span>
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># ä¼˜ç‚¹:</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># 1) wget è‡ªå¸¦çš„è¿›åº¦æ¡æ–¹ä¾¿æŸ¥çœ‹ä¸‹è½½è¿›åº¦</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># 2) å¯ä½¿ç”¨é­”æ³•åŠ é€Ÿ, è¿™é‡Œä¸èµ˜è¿°</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># ç¼ºç‚¹: ä¸‹è½½å¤šä¸ª .sra æ–‡ä»¶æ—¶éœ€è¦å¤šå¼€æˆ–è€…é¢å¤–æ“ä½œ</span>
</span></span></code></pre></div><h4 id="æ–¹æ³•ä¸‰-prefetch--ascp-åŠ é€Ÿä¸‹è½½">æ–¹æ³•ä¸‰ (prefetch + ascp åŠ é€Ÿä¸‹è½½)</h4>
<ul>
<li>å¦‚æœä½ ä½¿ç”¨ <code>anaconda</code>:</li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span><span style="color:#75715e"># å‡†å¤‡å·¥ä½œ, åªéœ€è¦åˆæ¬¡ä½¿ç”¨æ—¶èµ°ä¸€éæµç¨‹, ä»¥ååªè¦ &#39;å¼€å§‹ä¸‹è½½&#39; é‚£æ­¥å°±å¯ä»¥äº†</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># é¦–å…ˆ, ç¡®ä¿æ‰€åœ¨è™šæ‹Ÿç¯å¢ƒä¸º sra-tools</span>
</span></span><span style="display:flex;"><span>conda activate sra-tools
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æ¥ç€, å®‰è£… aspera-connect</span>
</span></span><span style="display:flex;"><span>conda install -c rpetit3 aspera-connect <span style="color:#75715e"># ç­‰å¾…ä¸€ä¼šå„¿åè¾“å…¥ y , ç„¶åå›è½¦</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æœå¯»ä½ çš„ç§é’¥æ–‡ä»¶, å¹¶æ·»åŠ åˆ°ç¯å¢ƒå˜é‡</span>
</span></span><span style="display:flex;"><span>echo <span style="color:#e6db74">&#34;ascp_key=</span><span style="color:#66d9ef">$(</span>find ~/anaconda3/envs/sra-tools/ -name asperaweb_id_dsa.openssh<span style="color:#66d9ef">)</span><span style="color:#e6db74">&#34;</span> &gt;&gt; ~/.bashrc;
</span></span><span style="display:flex;"><span>source ~/.bashrc
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æ£€æŸ¥æ˜¯å¦æˆåŠŸæ‰¾åˆ°ç§é’¥</span>
</span></span><span style="display:flex;"><span>echo $ascp_key 
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æˆ‘çš„è¾“å‡º:</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># /home/saturn/anaconda3/envs/sra-tools/opt/aspera/connect/etc/asperaweb_id_dsa.openssh</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æ²¡æœ‰è¾“å‡ºçš„è¯è¯´æ˜æœ‰é—®é¢˜, éœ€è¦è‡ªè¡Œæ’æŸ¥</span>
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># å¼€å§‹ä¸‹è½½</span>
</span></span><span style="display:flex;"><span>prefetch -a <span style="color:#e6db74">&#34;</span><span style="color:#66d9ef">$(</span>which ascp<span style="color:#66d9ef">)</span><span style="color:#e6db74">|</span>$ascp_key<span style="color:#e6db74">&#34;</span> SRR1646444
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æ²¡é—®é¢˜çš„è¯ä¼šæ˜¾ç¤º:</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># 2024-10-14T13:14:20 prefetch.3.1.1: 1) Resolving &#39;SRR1646444&#39;...</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># 2024-10-14T13:14:23 prefetch.3.1.1: Current preference is set to retrieve SRA Normalized Format files with full base quality scores</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># 2024-10-14T13:14:24 prefetch.3.1.1: 1) Downloading &#39;SRR1646444&#39;...</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># 2024-10-14T13:14:24 prefetch.3.1.1:  SRA Normalized Format file is being retrieved</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># 2024-10-14T13:14:24 prefetch.3.1.1:  Downloading via HTTPS...</span>
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æ–‡ä»¶ SRR1646444.sra ä¿å­˜åœ¨å½“å‰ç›®å½•çš„ SRR1646444 æ–‡ä»¶å¤¹é‡Œé¢</span>
</span></span></code></pre></div><ul>
<li>å¦‚æœä½ ä½¿ç”¨ <code>micromamba</code>:</li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span><span style="color:#75715e"># å‡†å¤‡å·¥ä½œ, åªéœ€è¦åˆæ¬¡ä½¿ç”¨æ—¶èµ°ä¸€éæµç¨‹</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># é¦–å…ˆ, ç¡®ä¿æ‰€åœ¨è™šæ‹Ÿç¯å¢ƒä¸º sra-tools</span>
</span></span><span style="display:flex;"><span>micromamba activate sra-tools
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æ¥ç€, å®‰è£… aspera-connect</span>
</span></span><span style="display:flex;"><span>micromamba install -c rpetit3 aspera-connect <span style="color:#75715e"># ç­‰å¾…ä¸€ä¼šå„¿åè¾“å…¥ Y , ç„¶åå›è½¦</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æœå¯»ä½ çš„ç§é’¥æ–‡ä»¶, å¹¶æ·»åŠ åˆ°ç¯å¢ƒå˜é‡</span>
</span></span><span style="display:flex;"><span>echo <span style="color:#e6db74">&#34;ascp_key=</span><span style="color:#66d9ef">$(</span>find ~/micromamba/envs/sra-tools/ -name asperaweb_id_dsa.openssh<span style="color:#66d9ef">)</span><span style="color:#e6db74">&#34;</span> &gt;&gt; ~/.bashrc;
</span></span><span style="display:flex;"><span>source ~/.bashrc
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æ£€æŸ¥æ˜¯å¦æˆåŠŸæ‰¾åˆ°ç§é’¥</span>
</span></span><span style="display:flex;"><span>echo $ascp_key
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æˆ‘çš„è¾“å‡º:</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># /home/saturn/micromamba/envs/sra-tools/opt/aspera/connect/etc/asperaweb_id_dsa.openssh</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æ²¡æœ‰è¾“å‡ºçš„è¯è¯´æ˜æœ‰é—®é¢˜, éœ€è¦è‡ªè¡Œæ’æŸ¥</span>
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># å¼€å§‹ä¸‹è½½</span>
</span></span><span style="display:flex;"><span>prefetch -a <span style="color:#e6db74">&#34;</span><span style="color:#66d9ef">$(</span>which ascp<span style="color:#66d9ef">)</span><span style="color:#e6db74">|</span>$ascp_key<span style="color:#e6db74">&#34;</span> SRR1646444
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æ²¡é—®é¢˜çš„è¯ä¼šæ˜¾ç¤º:</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># 2024-10-14T13:14:20 prefetch.3.1.1: 1) Resolving &#39;SRR1646444&#39;...</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># 2024-10-14T13:14:23 prefetch.3.1.1: Current preference is set to retrieve SRA Normalized Format files with full base quality scores</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># 2024-10-14T13:14:24 prefetch.3.1.1: 1) Downloading &#39;SRR1646444&#39;...</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># 2024-10-14T13:14:24 prefetch.3.1.1:  SRA Normalized Format file is being retrieved</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># 2024-10-14T13:14:24 prefetch.3.1.1:  Downloading via HTTPS...</span>
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æ–‡ä»¶ SRR1646444.sra ä¿å­˜åœ¨å½“å‰ç›®å½•çš„ SRR1646444 æ–‡ä»¶å¤¹é‡Œé¢</span>
</span></span></code></pre></div><hr>
<h3 id="è½¬æ¢-sra-æ–‡ä»¶ä¸º-fastqgz">è½¬æ¢ <code>.sra</code> æ–‡ä»¶ä¸º <code>fastq.gz</code></h3>
<p>(å‡å¦‚æˆ‘æƒ³æŠŠå‰é¢ä¸‹è½½çš„ <code>SRR1646444.sra</code> è½¬æ¢æˆ <code>SRR1646444.fastq.gz</code>)</p>
<h4 id="ä½¿ç”¨-fastq-dump">ä½¿ç”¨ <code>fastq-dump</code></h4>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span>fastq-dump --gzip --split-files SRR1646444.sra
</span></span><span style="display:flex;"><span><span style="color:#75715e"># ä¼šåœ¨å½“å‰ç›®å½•ç”Ÿæˆ SRR1646444.fastq.gz</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># fastq-dump è¿˜æ”¯æŒé€šé…ç¬¦, å¯è½¬æ¢å¤šä¸ª .sra æ–‡ä»¶</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># ä¾‹: fastq-dump --gzip --split-files *.sra</span>
</span></span></code></pre></div><h4 id="ä½¿ç”¨-fasterq-dump">ä½¿ç”¨ <code>fasterq-dump</code></h4>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span><span style="color:#75715e"># è™½ç„¶ fasterq-dump åœ¨å¤„ç†é€Ÿåº¦ä¸Šæ¯” fastq-dumpå¿«, ä½†æ˜¯å¹¶æ²¡æœ‰æä¾› --gzip é€‰é¡¹ç”¨äºè¾“å‡º gzip å‹ç¼©åçš„ fastq æ–‡ä»¶</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># å¹³æ›¿æ–¹æ³•å¦‚ä¸‹:</span>
</span></span><span style="display:flex;"><span>fasterq-dump --split-files SRR1646444.sra -o /dev/stdout | gzip -c &gt; SRR1646444.fastq.gz
</span></span><span style="display:flex;"><span><span style="color:#75715e"># åŸç†: å…ˆå°† .sra æ–‡ä»¶è§£å‹åè¾“å‡ºåˆ°æ ‡å‡†è¾“å‡º, ç„¶åä½¿ç”¨ç®¡é“ç¬¦å¯¼å…¥åˆ° gzip å®æ—¶å‹ç¼©æˆ .fastq.gz æ–‡ä»¶</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># PS: æƒ³è¦è¿›ä¸€æ­¥æé€Ÿ, å¯ä»¥ä½¿ç”¨ pigz æ›¿æ¢ gzip -c (æ³¨æ„: pigz éœ€å®‰è£…! )</span>
</span></span></code></pre></div><hr>
<h4 id="é™„-é»˜è®¤å‚æ•°ä¸‹-fastq-dump-ä¸-fasterq-dump-çš„å¤„ç†é€Ÿåº¦å¯¹æ¯”">é™„: é»˜è®¤å‚æ•°ä¸‹ <code>fastq-dump</code> ä¸ <code>fasterq-dump</code> çš„å¤„ç†é€Ÿåº¦å¯¹æ¯”</h4>
<p>(æˆ‘ä½¿ç”¨ shell çš„æ˜¯ <code>fish</code> , è€Œé <code>bash</code> , å˜»å˜»)</p>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span><span style="color:#75715e"># fastq-dump (è€—æ—¶çº¦ 443 ç§’)</span>
</span></span><span style="display:flex;"><span>time fastq-dump --gzip --split-files SRR1646444
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Read 2631092 spots for SRR1646444</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Written 2631092 spots for SRR1646444</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># ________________________________________________________</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Executed in  443.97 secs    fish           external</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e">#    usr time   53.23 secs    0.00 micros   53.23 secs</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e">#    sys time    1.34 secs  914.00 micros    1.33 secs</span>
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># fasterq-dump (è€—æ—¶çº¦ 30 ç§’)</span>
</span></span><span style="display:flex;"><span>time fasterq-dump --split-files ./SRR1646444.sra -o /dev/stdout | gzip -c &gt; SRR1646444.fastq.gz.fasterq-dump
</span></span><span style="display:flex;"><span><span style="color:#75715e"># spots read      : 2,631,092</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># reads read      : 2,631,092</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># reads written   : 2,631,092</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># ________________________________________________________</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Executed in   30.31 secs    fish           external</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e">#    usr time   35.21 secs    0.78 millis   35.21 secs</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e">#    sys time    3.68 secs    1.17 millis    3.68 secs</span>
</span></span></code></pre></div><hr>
<h3 id="æŸ¥çœ‹-fastqgz-æ–‡ä»¶">æŸ¥çœ‹ <code>.fastq.gz</code> æ–‡ä»¶</h3>
<h4 id="æ–¹æ³•ä¸€-gunzip--less">æ–¹æ³•ä¸€ (gunzip + less)</h4>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span>gunzip -c SRR1646444.fastq.gz | less
</span></span></code></pre></div><h4 id="æ–¹æ³•äºŒ-zcat--less">æ–¹æ³•äºŒ (zcat + less)</h4>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span>zcat SRR1646444.fastq.gz | less
</span></span></code></pre></div><h4 id="æ–¹æ³•ä¸‰-zless">æ–¹æ³•ä¸‰ (zless)</h4>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span>zless SRR1646444.fastq.gz
</span></span><span style="display:flex;"><span><span style="color:#75715e"># &#34;Less is more.&#34;</span>
</span></span></code></pre></div><hr>
<h2 id="fastqc"><code>FastQC</code></h2>
<h3 id="ç”¨æ³•è§£æ">ç”¨æ³•è§£æ</h3>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span>fastqc <span style="color:#f92672">[</span>é€‰é¡¹<span style="color:#f92672">]</span> &lt;fastq_file1&gt; &lt;fastq_file2&gt; ...
</span></span></code></pre></div><h3 id="å¸¸ç”¨é€‰é¡¹">å¸¸ç”¨é€‰é¡¹</h3>
<ul>
<li><code>-o &lt;dir&gt;</code> : æŒ‡å®šè¾“å‡ºæ–‡ä»¶è·¯å¾„, é»˜è®¤ä¸ºå½“å‰è·¯å¾„</li>
<li><code>-t &lt;int&gt;</code> : æŒ‡å®šçº¿ç¨‹æ•°ç›®</li>
<li><code>-a &lt;filename&gt;</code> : æŒ‡å®š adapter æ–‡ä»¶</li>
</ul>
<h3 id="å®æˆ˜ä½¿ç”¨">å®æˆ˜ä½¿ç”¨</h3>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span>fastqc SRR1646444.fastq.gz
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># application/gzip</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Started analysis of SRR1646444.fastq.gz</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Approx 5% complete for SRR1646444.fastq.gz</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Approx 10% complete for SRR1646444.fastq.gz</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Approx 15% complete for SRR1646444.fastq.gz</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Approx 20% complete for SRR1646444.fastq.gz</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Approx 25% complete for SRR1646444.fastq.gz</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Approx 30% complete for SRR1646444.fastq.gz</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Approx 35% complete for SRR1646444.fastq.gz</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Approx 40% complete for SRR1646444.fastq.gz</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Approx 45% complete for SRR1646444.fastq.gz</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Approx 50% complete for SRR1646444.fastq.gz</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Approx 55% complete for SRR1646444.fastq.gz</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Approx 60% complete for SRR1646444.fastq.gz</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Approx 65% complete for SRR1646444.fastq.gz</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Approx 70% complete for SRR1646444.fastq.gz</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Approx 75% complete for SRR1646444.fastq.gz</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Approx 80% complete for SRR1646444.fastq.gz</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Approx 85% complete for SRR1646444.fastq.gz</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Approx 90% complete for SRR1646444.fastq.gz</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Approx 95% complete for SRR1646444.fastq.gz</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Analysis complete for SRR1646444.fastq.gz</span>
</span></span></code></pre></div><h3 id="è¾“å‡ºæ–‡ä»¶">è¾“å‡ºæ–‡ä»¶</h3>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span>ls *fastqc*
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># SRR1646444_fastqc.html    SRR1646444_fastqc.zip</span>
</span></span></code></pre></div><ul>
<li><code>.html æ–‡ä»¶</code> ï¼šåŒ…å«å¯è§†åŒ–çš„è´¨é‡æŠ¥å‘Šï¼Œå¯åœ¨æµè§ˆå™¨ä¸­æŸ¥çœ‹</li>
<li><code>.zip  æ–‡ä»¶</code> ï¼šåŒ…å«è¯¦ç»†çš„ç»Ÿè®¡æ•°æ®åŠå›¾è¡¨</li>
</ul>
<hr>
<h2 id="multiqc"><code>MultiQC</code></h2>
<h3 id="ç”¨æ³•è§£æ-1">ç”¨æ³•è§£æ</h3>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span>multiqc <span style="color:#f92672">[</span>é€‰é¡¹<span style="color:#f92672">]</span> &lt;data_directory&gt;
</span></span></code></pre></div><h3 id="å¸¸ç”¨é€‰é¡¹-1">å¸¸ç”¨é€‰é¡¹</h3>
<ul>
<li><code>-o &lt;dir&gt;</code> : æŒ‡å®šè¾“å‡ºæ–‡ä»¶è·¯å¾„, é»˜è®¤ä¸ºå½“å‰è·¯å¾„</li>
<li><code>-n &lt;name&gt;</code>ï¼šæŒ‡å®šç”ŸæˆæŠ¥å‘Šçš„æ–‡ä»¶åï¼Œ é»˜è®¤ä¸º <code>multiqc_report.html</code></li>
</ul>
<h3 id="å®æˆ˜ä½¿ç”¨-1">å®æˆ˜ä½¿ç”¨</h3>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span>multiqc /public/workspace/shaojf/Course/NGS/DataSets/Lab1/rawdata.fastqc.result/
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e">#   /// MultiQC ğŸ” | v1.17</span>
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># |           multiqc | Search path : /public/workspace/shaojf/Course/NGS/DataSets/Lab1/rawdata.fastqc.result</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># |         searching | â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â” 100% 36/36  </span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># |            fastqc | Found 18 reports</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># |           multiqc | Report      : multiqc_report.html</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># |           multiqc | Data        : multiqc_data</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># |           multiqc | MultiQC complete</span>
</span></span></code></pre></div><h3 id="è¾“å‡ºæ–‡ä»¶-1">è¾“å‡ºæ–‡ä»¶</h3>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span>ls *multiqc*
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># multiqc_report.html</span>
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># multiqc_data:</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># multiqc.log</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># multiqc_citations.txt</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># multiqc_data.json</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># multiqc_fastqc.txt</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># multiqc_general_stats.txt</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># multiqc_software_versions.txt</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># multiqc_sources.txt</span>
</span></span></code></pre></div><ul>
<li><code>multiqc_report.html</code> : ä¸»è¦çš„å¯è§†åŒ–æŠ¥å‘Šï¼ŒåŒ…å«äº†å¤šç§äº¤äº’å¼å›¾è¡¨ï¼Œå¯åœ¨æµè§ˆå™¨ä¸­æ‰“å¼€æŸ¥çœ‹</li>
<li><code>multiqc_data/</code> æ–‡ä»¶å¤¹ : ä¿å­˜äº†æ‰€æœ‰ç”¨äºç”ŸæˆæŠ¥å‘Šçš„åŸå§‹æ•°æ®æ–‡ä»¶å’Œä¸€äº›è¾…åŠ©ä¿¡æ¯</li>
</ul>
<hr>
<h2 id="ä»»åŠ¡">ä»»åŠ¡</h2>
<ul>
<li>æ•°æ®ï¼š <code>/public/workspace/shaojf/Course/NGS/DataSets/Lab1</code></li>
</ul>
<ol>
<li>ï¼ˆé€‰åšï¼‰è‡ªè¡Œå®‰è£… <code>sratools</code>ã€<code>FastQC</code>ã€<code>MultiQC</code></li>
<li><code>SRR1646444.sra</code> è½¬æˆ <code>.fastq.gz</code> æ ¼å¼, å¹¶ç”¨ <code>FastQC</code> è¯„ä¼°</li>
<li>ç”¨multiQCç”Ÿæˆ <code>rawdata.fastqc.result</code> ç›®å½•ä¸‹çš„æ±‡æ€»QCæŠ¥å‘Š</li>
</ol>
<p>( ç»“æœæ–‡ä»¶å‡ä¿å­˜è‡³ <code>~/myNGS/Lab1/</code> æ–‡ä»¶å¤¹)</p>
<h2 id="å‚è€ƒä»£ç ">å‚è€ƒä»£ç </h2>
<h3 id="1-å®‰è£…ç›¸å…³è½¯ä»¶">1. å®‰è£…ç›¸å…³è½¯ä»¶</h3>
<ul>
<li>ä¸èµ˜è¿°, è¯¦è§ä»¥ä¸Šå†…å®¹</li>
</ul>
<h3 id="2-æ ¼å¼è½¬æ¢ä¸è¯„ä¼°">2. æ ¼å¼è½¬æ¢ä¸è¯„ä¼°</h3>
<ul>
<li>ç¡®ä¿æ–‡ä»¶å¤¹å·²ç»å»ºå¥½</li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span>mkdir -p ~/myNGS/Lab1
</span></span></code></pre></div><ul>
<li>è¿›å…¥æ–‡ä»¶å¤¹</li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span>cd ~/myNGS/Lab1
</span></span></code></pre></div><ul>
<li>å°† .sra æ•°æ®è½¯é“¾æ¥åˆ° Lab1 æ–‡ä»¶å¤¹å†…, è¿™æ ·å°±ä¸ç”¨è‡ªå·±ä¸‹è½½å’¯ï½</li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span>ln -s /public/workspace/shaojf/Course/NGS/DataSets/Lab1/SRR1646444.sra .
</span></span></code></pre></div><ul>
<li>è½¬æ¢ .sra æ–‡ä»¶ä¸º .fastq.gz æ ¼å¼ (ä¸¤ç§æ–¹æ³•ä»»é€‰ä¸€ç§)</li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span><span style="color:#75715e"># æ–¹æ³•ä¸€ï¼Œä½¿ç”¨ fastq-dump, è¾ƒæ…¢</span>
</span></span><span style="display:flex;"><span>fastq-dump --split-files SRR1646444.sra
</span></span><span style="display:flex;"><span>gzip SRR1646444.fastq
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># æ–¹æ³•äºŒï¼Œä½¿ç”¨ fasterq-dump, è¾ƒå¿«</span>
</span></span><span style="display:flex;"><span>fasterq-dump --split-files SRR1646444.sra -o /dev/stdout | gzip -c &gt; SRR1646444.fastq.gz
</span></span></code></pre></div><ul>
<li>å»ºç«‹ä¸¤ä¸ªæ–‡ä»¶å¤¹, ç”¨äºä¿å­˜ä½¿ç”¨ä¸åŒ adapter çš„ç»“æœæ–‡ä»¶</li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span>mkdir default.qc specified.qc
</span></span></code></pre></div><ul>
<li>ä½¿ç”¨ <code>FastQC</code> è¿›è¡Œè¯„ä¼°</li>
</ul>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span><span style="color:#75715e"># ç¬¬ä¸€æ¬¡è¯„ä¼°: ä½¿ç”¨é»˜è®¤ adapter, è¾“å‡ºç»“æœåˆ° default.qc æ–‡ä»¶å¤¹</span>
</span></span><span style="display:flex;"><span>fastqc -o default.qc SRR1646444.fastq.gz
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># ç¬¬äºŒæ¬¡è¯„ä¼°: æŒ‡å®šç‰¹å®š adapter, è¾“å‡ºç»“æœåˆ° specified.qc æ–‡ä»¶å¤¹</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># å°† adapter æ–‡ä»¶è½¯é“¾æ¥åˆ°ä½ çš„ Lab1 é‡Œé¢</span>
</span></span><span style="display:flex;"><span>ln -s /public/workspace/shaojf/Course/NGS/Tools/FastQC/Configuration/adapter_list.txt .
</span></span><span style="display:flex;"><span><span style="color:#75715e"># ä½¿ç”¨ -a å‚æ•°æŒ‡å®š adapter æ–‡ä»¶</span>
</span></span><span style="display:flex;"><span>fastqc -o specified.qc -a adapter_list.txt SRR1646444.fastq.gz
</span></span></code></pre></div><h3 id="3-æ±‡æ€»qcæŠ¥å‘Š">3. æ±‡æ€»QCæŠ¥å‘Š</h3>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-shell" data-lang="shell"><span style="display:flex;"><span><span style="color:#75715e"># å°†å­˜æ”¾å¤šä¸ªQCæŠ¥å‘Šçš„æ–‡ä»¶å¤¹è½¯é“¾æ¥åˆ°ä½ çš„ Lab1 é‡Œé¢</span>
</span></span><span style="display:flex;"><span>ln -s /public/workspace/shaojf/Course/NGS/DataSets/Lab1/rawdata.fastqc.result/ .
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># ä½¿ç”¨MultiQCç”Ÿæˆæ±‡æ€»æŠ¥å‘Š</span>
</span></span><span style="display:flex;"><span>multiqc rawdata.fastqc.result/
</span></span></code></pre></div>
        </div>

        
        
        <div class="article-toc" >
            <h3>Contents</h3>
            <nav id="TableOfContents">
  <ul>
    <li><a href="#æ•°æ®åº“ç½‘ç«™">æ•°æ®åº“ç½‘ç«™</a></li>
    <li><a href="#æ•°æ®ä¸‹è½½å·¥å…·">æ•°æ®ä¸‹è½½å·¥å…·</a></li>
    <li><a href="#è´¨é‡è¯„ä¼°å·¥å…·">è´¨é‡è¯„ä¼°å·¥å…·</a></li>
    <li><a href="#å®‰è£…-sra-toolkit">å®‰è£… <code>SRA Toolkit</code></a>
      <ul>
        <li><a href="#ä½¿ç”¨-anaconda-å®‰è£…">ä½¿ç”¨ <code>anaconda</code> å®‰è£…</a></li>
        <li><a href="#ä½¿ç”¨-micromamba-å®‰è£…">ä½¿ç”¨ <code>micromamba</code> å®‰è£…</a></li>
      </ul>
    </li>
    <li><a href="#å®‰è£…-fastqc">å®‰è£… <code>FastQC</code></a>
      <ul>
        <li><a href="#ä½¿ç”¨-anaconda-å®‰è£…-1">ä½¿ç”¨ <code>anaconda</code> å®‰è£…</a></li>
        <li><a href="#ä½¿ç”¨-micromamba-å®‰è£…-1">ä½¿ç”¨ <code>micromamba</code> å®‰è£…</a></li>
      </ul>
    </li>
    <li><a href="#å®‰è£…-multiqc">å®‰è£… <code>MultiQC</code></a>
      <ul>
        <li><a href="#ä½¿ç”¨-anaconda-å®‰è£…-2">ä½¿ç”¨ <code>anaconda</code> å®‰è£…</a></li>
        <li><a href="#ä½¿ç”¨-micromamba-å®‰è£…-2">ä½¿ç”¨ <code>micromamba</code> å®‰è£…</a></li>
      </ul>
    </li>
    <li><a href="#ä½¿ç”¨æ•™ç¨‹">ä½¿ç”¨æ•™ç¨‹</a></li>
    <li><a href="#sra-toolkit"><code>SRA Toolkit</code></a>
      <ul>
        <li><a href="#ä¸‹è½½-sra-æ–‡ä»¶">ä¸‹è½½ <code>.sra</code> æ–‡ä»¶</a></li>
        <li><a href="#è½¬æ¢-sra-æ–‡ä»¶ä¸º-fastqgz">è½¬æ¢ <code>.sra</code> æ–‡ä»¶ä¸º <code>fastq.gz</code></a></li>
        <li><a href="#æŸ¥çœ‹-fastqgz-æ–‡ä»¶">æŸ¥çœ‹ <code>.fastq.gz</code> æ–‡ä»¶</a></li>
      </ul>
    </li>
    <li><a href="#fastqc"><code>FastQC</code></a>
      <ul>
        <li><a href="#ç”¨æ³•è§£æ">ç”¨æ³•è§£æ</a></li>
        <li><a href="#å¸¸ç”¨é€‰é¡¹">å¸¸ç”¨é€‰é¡¹</a></li>
        <li><a href="#å®æˆ˜ä½¿ç”¨">å®æˆ˜ä½¿ç”¨</a></li>
        <li><a href="#è¾“å‡ºæ–‡ä»¶">è¾“å‡ºæ–‡ä»¶</a></li>
      </ul>
    </li>
    <li><a href="#multiqc"><code>MultiQC</code></a>
      <ul>
        <li><a href="#ç”¨æ³•è§£æ-1">ç”¨æ³•è§£æ</a></li>
        <li><a href="#å¸¸ç”¨é€‰é¡¹-1">å¸¸ç”¨é€‰é¡¹</a></li>
        <li><a href="#å®æˆ˜ä½¿ç”¨-1">å®æˆ˜ä½¿ç”¨</a></li>
        <li><a href="#è¾“å‡ºæ–‡ä»¶-1">è¾“å‡ºæ–‡ä»¶</a></li>
      </ul>
    </li>
    <li><a href="#ä»»åŠ¡">ä»»åŠ¡</a></li>
    <li><a href="#å‚è€ƒä»£ç ">å‚è€ƒä»£ç </a>
      <ul>
        <li><a href="#1-å®‰è£…ç›¸å…³è½¯ä»¶">1. å®‰è£…ç›¸å…³è½¯ä»¶</a></li>
        <li><a href="#2-æ ¼å¼è½¬æ¢ä¸è¯„ä¼°">2. æ ¼å¼è½¬æ¢ä¸è¯„ä¼°</a></li>
        <li><a href="#3-æ±‡æ€»qcæŠ¥å‘Š">3. æ±‡æ€»QCæŠ¥å‘Š</a></li>
      </ul>
    </li>
  </ul>
</nav>
        </div>
        
        

        


        
    </div>
    
<nav id="article-nav">
    
    <a href="/ngs/ngs_lab2/" id="article-nav-newer" class="article-nav-link-wrap">
        <div class="article-nav-title"><span>&lt;</span>&nbsp;
            NGS_lab2
        </div>
    </a>
    
    
    <a href="/ngs/ngs_lab0/" id="article-nav-older" class="article-nav-link-wrap">
        <div class="article-nav-title">NGS_lab0&nbsp;<span>&gt;</span></div>
    </a>
    
</nav>


</article>

        
    </section>
    <footer id="footer">
    <div class="outer">
        <div id="footer-info" class="inner">
            &copy; 2024 Meteor03&#39;s Oxalis Garden
            <br />
            Powered by <a href="https://gohugo.io" target="_blank">Hugo</a> with theme <a href="https://github.com/carsonip/hugo-theme-minos" target="_blank">Minos</a>
        </div>
    </div>

    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.9.0/styles/tomorrow-night.min.css" integrity="sha256-2wL88NKUqvJi/ExflDzkzUumjUM73mcK2gBvBBeLvTk=" crossorigin="anonymous" />
    <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.9.0/highlight.min.js" integrity="sha256-KbfTjB0WZ8vvXngdpJGY3Yp3xKk+tttbqClO11anCIU=" crossorigin="anonymous"></script>
    <script>hljs.initHighlightingOnLoad();</script>

    
    <script>
        document.getElementById('main-nav-toggle').addEventListener('click', function () {
            var header = document.getElementById('header');
            if (header.classList.contains('mobile-on')) {
                header.classList.remove('mobile-on');
            } else {
                header.classList.add('mobile-on');
            }
        });
    </script>
</footer>

</div>
</body>
</html>
